### PR TITLE
Add selected state for sources

### PR DESCRIPTION
**Pull request by jasonLaster**
Here's what it looks like 

gif

### CODE REVIEW

File: js/components/Sources.js [hunk 1]
```diff
@@ -1,27 +1,65 @@
...
+/**
+ * Takes a sources object indexed by actor and
+ * returns a sources object indexed by source domain.
+ *
+ * @returns Object
+ */
+function groupSourcesByDomain(sources) {
+  return groupBy(values(sources), source => {
+    if (!source.url) {
+      return source.introductionType;
```
Comment (jasonLaster) [inline js/components/Sources.js]: I'm not really sure what introductionType is doing, but this is what we were doing before, so i kept it

File: js/utils.js [hunk 2]
```diff
@@ -73,12 +73,35 @@ function deleteIn(destObj, path) {
   return setIn(destObj, objPath, obj.without(propName));
 }
 
+function groupBy(array, predicate) {
```
Comment (jasonLaster) [inline js/utils.js]: Related to issue #31. We can add a couple small utility functions that DRY up our code a lot.

File: js/components/Sources.js (hunk 1 above)
Comment (jlongster) [inline js/components/Sources.js]: `introductionType` comes from the JS engine and tell you what kind of source it is (eval, new Function, script tag, etc). Many scripts don't have a URL if they are eval'ed. You see this on any complex site; the list of sources includes things like `Function`, `eval`, etc. Eventually we will want to treat this differently.

**Comment by jlongster** [comment]
I'm about to land the immutable.js integration. You don't need your extra utility functions then. Do you want to rebase this on top of my work?

**Comment by clarkbw** [comment]
If we don't merge this today, nobody gets any 🍻 
😄

**Comment by jasonLaster** [comment]
merged! 
:beer: please

**Comment by jlongster** [comment]
I'll be checking my mail tomorrow.

**Pull request closed without merging**